container/lxd/manager.go

```diff
@@ -289,6 +291,8 @@ func (m *containerManager) networkDevicesFromConfig(netConfig *container.Network
 
 // MaybeWriteLXDProfile implements container.LXDProfileManager.
 func (m *containerManager) MaybeWriteLXDProfile(pName string, put *charm.LXDProfile) error {
+	m.profileMutex.Lock()
```

Comment:
drive by to fix: 0        down                  pending        bionic      cannot write lxd profiles: Error inserting juju-default-lxd-profile-0 into database: The profile already exists